pkg/util/selectors/bimultimap.go

```diff
@@ -0,0 +1,380 @@
+/*
```

Comment:
Was there any discussion about adding this to `pkg/controller/util/selectors` instead?  It's much more likely to show up in that context and `pkg/util` has problematic implications to overall cohesion.

Reply:
No, we did not discuss that. But I would agree that having more narrow context for that lib makes more sense.

Reply:
Fixed.